specification/search/data-plane/Azure.Search/preview/2024-11-01-preview/searchindex.json

```diff
@@ -0,0 +1,3220 @@
+                  "name": "Extractive",
+                  "description": "Extracts captions from the matching documents that contain passages relevant to the search query."
+                }
+              ]
+            },
+            "description": "This parameter is only valid if the query type is `semantic`. If set, the query returns captions extracted from key passages in the highest ranked documents. When Captions is set to `extractive`, highlighting is enabled by default, and can be configured by appending the pipe character `|` followed by the `highlight-<true/false>` option, such as `extractive|highlight-true`. Defaults to `None`. The maximum character length of captions can be configured by appending the pipe character '|' followed by the 'count-<number of maximum character length>', such as 'extractive|maxcharlength-600'.",
+            "x-ms-parameter-grouping": {
+              "name": "SearchOptions"
+            }
+          },
+          {
+            "name": "queryRewrites",
+            "in": "query",
+            "type": "string",
+            "enum": [
+              "none",
+              "generative"
+            ],
+            "x-ms-enum": {
+              "name": "QueryRewritesType",
+              "modelAsString": true,
+              "values": [
+                {
+                  "value": "none",
+                  "name": "None",
+                  "description": "Do not generate additional query rewrites for this query."
+                },
+                {
+                  "value": "generative",
+                  "name": "Generative",
+                  "description": "Generate alternative query terms to increase the recall of a search request."
+                }
+              ]
+            },
+            "description": "When QueryRewrites is set to `generative`, the query terms are sent to a generate model which will produce 10 (default) rewrites to help increase the recall of the request. The requested count can be configured by appending the pipe character `|` followed by the `count-<number of rewrites>` option, such as `generative|count-3`. Defaults to `None`. This parameter is only valid if the query type is `semantic`.",
+            "x-ms-parameter-grouping": {
+              "name": "SearchOptions"
+            }
+          },
+          {
+            "name": "semanticQuery",
+            "in": "query",
+            "type": "string",
+            "description": "Allows setting a separate search query that will be solely used for semantic reranking, semantic captions and semantic answers. Is useful for scenarios where there is a need to use different queries between the base retrieval and ranking phase, and the L2 semantic phase.",
+            "x-ms-parameter-grouping": {
+              "name": "SearchOptions"
+            }
+          },
+          {
+            "name": "debug",
+            "in": "query",
+            "type": "string",
+            "enum": [
+              "disabled",
```

Comment:
As mentioned in another comment elsewhere, `all` is supported and we should look into adding that as an option here.